--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,8 +151,6 @@ endif()
 # Target and link #
 ###################
 
-set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib; ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-
 # xeus-cling sources
 set(XEUS_CLING_SRC
     src/xinput.hpp
@@ -205,21 +203,6 @@ target_link_libraries(xcpp PRIVATE xeus-
 
 set(LLVM_NO_DEAD_STRIP 1)
 
-if (APPLE)
-    set_target_properties(xcpp xeus-cling PROPERTIES
-        MACOSX_RPATH ON
-    )
-else()
-    set_target_properties(xcpp xeus-cling PROPERTIES
-        BUILD_WITH_INSTALL_RPATH 1
-        SKIP_BUILD_RPATH FALSE
-    )
-endif()
-
-set_target_properties(xcpp xeus-cling PROPERTIES
-    INSTALL_RPATH_USE_LINK_PATH TRUE
-)
-
 target_include_directories(xeus-cling
                            PUBLIC
                            $<BUILD_INTERFACE:${XEUS_CLING_INCLUDE_DIR}>
